File: solutions/cpp/rotational-cipher/3/rotational_cipher.cpp
#include "rotational_cipher.h"

#include <cctype>
namespace rotational_cipher {

    std::string rotate(const std::string& input, int n) {
        n %= 26;
        if (n < 0) {
            n += 26;
        }
        std::string result{};
        for (auto&& ch: input) {
            if (std::isupper(ch)) {
                result.push_back(std::isupper(ch + n) ? ch + n : ch + n - 'Z' - 1 + 'A');
            } else if (std::islower(ch)) {
                result.push_back(std::islower(ch + n) ? ch + n : ch + n - 'z' - 1 + 'a');
            } else {
                result.push_back(ch);
            }
        }
        return result;
    }

}  // namespace rotational_cipher

File: solutions/python/rotational-cipher/1/rotational_cipher.py
def rotate(text, key):
    return ''.join((
        chr((ord(ch) - ord('A') + key) % 26 + ord('A'))
        if ch.isupper() else
        chr((ord(ch) - ord('a') + key) % 26 + ord('a'))
        if ch.islower() else ch
        for ch in text
    ))